lib/types/index.ts

```diff
@@ -1,5 +1,19 @@
-export interface Config {
-  msg: string;
-  count: number;
-  interval?: number;
+import { ALL_CHANNELS, ALL_DB_ENGINES } from '../consts';
+
+export type DBEngine = (typeof ALL_DB_ENGINES)[number];
+
+export type AuthChannel = (typeof ALL_CHANNELS)[number];
+
+export interface LocalOptions {
+  dbEngine?: DBEngine;
+}
+
+export interface WardenConfig {
+  channels?: AuthChannel[] | AuthChannel | string;
```

Comment:
channels is having flexibility but every time you have to use it, it becomes a headache , refactor
study Single Responsibility Principle